com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/BuildCloudDeployModelStep.java

```diff
@@ -240,22 +242,27 @@ protected ServiceKeysCloudModelBuilder getServiceKeysCloudModelBuilder(ProcessCo
     private List<String> getDomainsFromApps(ProcessContext context, DeploymentDescriptor descriptor,
                                             ApplicationCloudModelBuilder applicationCloudModelBuilder, List<? extends Module> modules,
                                             ModuleToDeployHelper moduleToDeployHelper) {
-        String defaultDomain = (String) descriptor.getParameters()
-                                                  .get(SupportedParameters.DEFAULT_DOMAIN);
-
         Set<String> domains = new TreeSet<>();
         for (Module module : modules) {
             if (!moduleToDeployHelper.isApplication(module)) {
                 continue;
             }
             ParametersChainBuilder parametersChainBuilder = new ParametersChainBuilder(context.getVariable(Variables.COMPLETE_DEPLOYMENT_DESCRIPTOR));
-            List<String> appDomains = applicationCloudModelBuilder.getApplicationDomains(parametersChainBuilder.buildModuleChain(module.getName()),
-                                                                                         module);
+            List<Map<String, Object>> paramsList = parametersChainBuilder.buildModuleChain(module.getName());
+
+            boolean noRoute = (Boolean) PropertiesUtil.getPropertyValue(paramsList, SupportedParameters.NO_ROUTE, false);

```

Comment:
This could be throw class cast exception if the user set non boolean value for "no-route" parameter. getProprtyValue method should be refactored to pass what class type you expect and throw exception if doesn't match to the required class type. If this refactor requires a lot of change, you can skip it but track BI in our jira or look for existing one and increase the prio.